src/routers/StackRouter.js

```diff
@@ -100,22 +100,41 @@ export default (routeConfigs, stackConfig = {}) => {
       // Set up the initial state if needed
       if (!state) {
         let route = {};
+        const childRouter = childRouters[action.routeName];
         if (
           action.type === NavigationActions.NAVIGATE &&
-          childRouters[action.routeName] !== undefined
+          childRouter !== undefined
         ) {
+          if (childRouter === null) {
+            return {
+              isTransitioning: false,
+              index: 0,
+              routes: [
+                {
+                  ...action,
```

Comment:
rather than putting in the whole action and overriding type, can we instead just explicitly use the routeName and params from the action?

Also we should use the key from the action, if it is provided

Reply:
👍

Reply:
Based on a conversation @ericvicenti and I had on Slack, it should always be using `action.key` if available, not just in this specific place. PR updated to reflect that.